packages/fee-estimation/src/estimateFees.ts

```diff
@@ -0,0 +1,363 @@
+import {
+  gasPriceOracleABI,
+  gasPriceOracleAddress,
+} from '@eth-optimism/contracts-ts'
+import {
+  getContract,
+  createPublicClient,
+  http,
+  BlockTag,
+  Address,
+  EstimateGasParameters,
+  serializeTransaction,
+  encodeFunctionData,
+  EncodeFunctionDataParameters,
+  TransactionSerializableEIP1559,
+  TransactionSerializedEIP1559,
+  stringify,
+} from 'viem'
+import * as chains from 'viem/chains'
+import { PublicClient } from 'wagmi'
+import { Abi } from 'abitype'
+import { writeFileSync } from 'fs'
+
+/**
+ * Bytes type representing a hex string with a 0x prefix
+ * @typedef {`0x${string}`} Bytes
+ */
+export type Bytes = `0x${string}`
+
+/**
+ * Options to query a specific block
+ */
+type BlockOptions = {
+  /**
+   * Block number to query from
+   */
+  blockNumber?: bigint
+  /**
+   * Block tag to query from
+   */
+  blockTag?: BlockTag
+}
+
+const knownChains = [chains.optimism.id, chains.goerli.id, chains.baseGoerli.id]
```

Comment:
should we expand this to base mainnet as well as zora

Reply:
Not mainnet because it's only relavent to L2 but zora for sure. I think I didn't add it here only because wagmi didn't support zora (yet) but will double check

Reply:
Zora has been added after updating viem to latest version!